docs/trex_xss_guidance.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Cross-Site Scripting and Extension Security
+layout: docs
+---
+
+As a developer, there are precautions and best practices to follow to developing secure extensions. A dashboard extension is a web application that runs inside the Tableau dashboard, and as a web application, your dashboard extension could be vulnerable to cross-site scripting (XSS). Cross-site scripting is an attack where a malicious user injects client-side code (typically, JavaScript) to execute in your web application. When other users load the web pages of your extension, the attacker's scripts can run, potentially stealing information or redirecting the browser to another web page.
+
+
+**In this section**
+
+* TOC
+{:toc}
+
+
+
+---
+
+## Basic steps toward preventing XSS
+
+If your dashboard extension accepts text input from users through forms or text boxes, you will want to ensure that you aren't accidentally introducing vulnerabilities. The best prevention for XSS attacks is to validate and encode any user input. There is a lot of good information available on the web for developers about how to protect web applications from attacks. For example, see  [Cross Site Scripting Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet){:target="_blank"} from the Open Web Application Security Project (OWASP).
+
+#### Key points:
+
+* Always validate user-provided text input into your HTML pages.
+  If your extension uses forms or text boxes to collect user input, be sure to validate and verify that the input conforms to what you expect. Use regular expressions to verify the input. 
+* Use HTML entity encoding and escape the characters `&`, `<`, `>`, `"`, `'`, and `/`.  For example, use entity names or numbers `&gt;` or `&#62;` for the greater than character `>`.
+* If your extension makes use URIs as input, use JavaScript methods, such as `encodeURI` and `encodeURIComponent` for encoding.
```

Comment:
"If your extension makes use _of_ URIs" (You are missing "of")

Reply:
Thanks!